### DISCUSSION TITLE
`consistentTypeDefintions`, @typescript-eslint/consistent-type-definitions

### DISCUSSION THREAD

**Discussion by jpike88** [rome/tools#4360] [Suggestions]
Often developers will write types and interfaces in interchangeable ways:

```
// type alias
type T1 = {
  a: string;
  b: number;
};

// interface keyword
interface T2 {
  a: string;
  b: number;
}
```

It's more consistent to require the above type `T1` to be defined as an interface as it 'looks' like one.

https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/consistent-type-definitions.md

**Comment by ematipico** [comment]
I would consider this proposal a very low priority. And not recommended.

Rome had a rule called `useInterface` but we removed because people can use whatever they want.

I would argue, instead, that it's better to create `type` by default, and `interface` when needed.

**Comment by Conaclos** [comment]
The TypeScript team advocates the use of `interface` over `type`. At the beginning of TypeScript `type` was poorly supported. However, it is no longer the case. The community use the two syntaxes.

I personally prefer `type` because it does not allow unsafe declaration merging.

I pretty like how Flow distinguishes `interface` and `type`:

- `type` for exact (no extra properties) literal objects
- `interfaces` for inexact literal objects, class's `implements`

Thus, in my view these two syntaxes communicate distinct intentions and programming paradigms. Both should be allowed. I think we should not implement this rule.

**Comment by jpike88** [comment]
I see your feedback and actually agree with you. My thinking is from an older time, it’s not relevant anymore. Closing
